File: ApiCredito2-main/GestionIntApi/DTO/NotificacionDTO.cs
namespace GestionIntApi.DTO
{
    public class NotificacionDTO
    {
        public int Id { get; set; }
        public int ClienteId { get; set; }
        public string Tipo { get; set; }
        public string Mensaje { get; set; }
        public DateTime Fecha { get; set; }
        public bool Leida { get; set; } = false;
    }
}

File: AppMovilGestionCreditos1/lib/models/notificacion_dto.dart
class NotificacionDTO {
  int id;
  int clienteId;
  String tipo;
  String mensaje;
  DateTime fecha;
  bool leida;

  NotificacionDTO({
    this.id = 0,
    required this.clienteId,
    required this.tipo,
    required this.mensaje,
    required this.fecha,
    this.leida = false,
  });

  factory NotificacionDTO.fromJson(Map<String, dynamic> json) {
    return NotificacionDTO(
      id: json['id'] ?? 0,
      clienteId: json['clienteId'] ?? 0,
      tipo: json['tipo'] ?? '',
      mensaje: json['mensaje'] ?? '',
      fecha: DateTime.parse(json['fecha']),
      leida: json['leida'] ?? false,
    );
  }

  Map<String, dynamic> toJson() => {
    'Id': id,
    'ClienteId': clienteId,
    'Tipo': tipo,
    'Mensaje': mensaje,
    'Fecha': fecha.toIso8601String(),
    'Leida': leida,
  };
}